src/Exception/OAuthServerException.php

```diff
@@ -334,13 +334,12 @@ public function getHttpHeaders()
         // respond with an HTTP 401 (Unauthorized) status code and
         // include the "WWW-Authenticate" response header field
         // matching the authentication scheme used by the client.
-        // @codeCoverageIgnoreStart
-        if ($this->errorType === 'invalid_client' && $this->serverRequest->hasHeader('Authorization') === true) {
+        if ($this->errorType === 'invalid_client' && $this->requestHasAuthorizationHeader()) {
```

Comment:
```suggestion
        if ($this->errorType === 'invalid_client' && !empty($this->serverRequest->getHeader('Authorization'))) {
```

We could remove the function and replace with the line change above I think

Reply:
I don't think so. The case we're trying to detect has a *non-empty* array containing a single empty string.

It would need to be something more like;

```patch
-        if ($this->errorType === 'invalid_client' && $this->requestHasAuthorizationHeader()) {
+        if ($this->errorType === 'invalid_client' && !empty($this->serverRequest->getHeader('Authorization')) && !empty($this->serverRequest->getHeader('Authorization')[0])) {
```

I think you suggested putting it out in a separate function on https://github.com/thephpleague/oauth2-server/issues/1162 because the above gets a bit unwieldy and didn't leave a good place to include an explanatory comment for this case.

Likely irrelevant, but if the code didn't need to be written to support earlier versions of PHP, it could be written more tersely (cryptically?) along these lines;

```php
$headers = $this->serverRequest->getHeaders();

if ($this->errorType === 'invalid_client' && !empty($headers['Authorization'][0] ?? '')) {
```

Reply:
Ahh sorry. I thoght it returned an empty array. Cheers for the explanation. Makes sense